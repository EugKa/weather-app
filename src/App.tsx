import React from 'react';
import { Route, Switch } from 'react-router-dom';
import './App.scss';
import { Header, NotFound } from './components';
import { MainPage, CityInfo } from './pages';

function App() {
  return (
    <div className="App">
      <Header/>
      <Switch>
        <Route exact component={MainPage} path='/'/>
        <Route component={CityInfo} path='/city/:name'/>
        <Route component={NotFound}/>
      </Switch>
    </div>
  );
}

export default App;
