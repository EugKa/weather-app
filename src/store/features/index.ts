export * from './current-weather/current-weather-slice';
export * from './hourecast-weather/housecast-weather-slice'