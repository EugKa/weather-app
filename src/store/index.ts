import { configureStore, combineReducers } from '@reduxjs/toolkit';

import CurrentWeatherData from './features/current-weather/current-weather-slice';
import HourCastWeatherSlice from './features/hourecast-weather/housecast-weather-slice';

export const rootReducer = combineReducers({
  CurrentWeatherData,
  HourCastWeatherSlice
})

export const store = configureStore({
  reducer: rootReducer,
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch