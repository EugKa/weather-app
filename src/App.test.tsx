import React from 'react';
import App from './App';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from './store';
import { shallow } from 'enzyme';


test('renders the component', () => {
  const component = shallow(
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>);
  expect(component).toMatchSnapshot();
}); 

test('renders App component', () => {
  const component = shallow(
      <App />
);
  expect(component).toMatchSnapshot();
}); 


