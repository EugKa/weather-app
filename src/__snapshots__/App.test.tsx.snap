// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component 1`] = `
<div
  className="App"
>
  <Header />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/city/:name"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;

exports[`renders the component 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <BrowserRouter>
    <App />
  </BrowserRouter>
</ContextProvider>
`;
